proj/src/main/webapp/script.js

```diff
@@ -134,16 +134,15 @@ function initRef() {
     // create database reference
     const dbRefObject = firebase.database().ref(PATH);
```

Comment:
can you call getPath here instead of using a constant.

Reply:
Maybe something like `var tag; path=getPath(tag,CHAT_ID); const dbRefObject = firebase.database().ref(path);` That way later you just need to get tag and chat_id from some client side cached data based on what the user clicked on to get to the page.